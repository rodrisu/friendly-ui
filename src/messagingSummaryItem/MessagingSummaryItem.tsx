import React from 'react'
import cc from 'classcat'

import { color } from '../_utils/branding'
import { Avatar } from '../avatar'
import { NormalizeProps } from '../layout/layoutNormalizer'
import { TextDisplayType } from '../text'
import { StyledMessagingSummaryItem } from './MessagingSummaryItem.style'

export type MessagingSummaryItemProps = NormalizeProps &
  Readonly<{
    className?: string
    url: string | JSX.Element
    pictureUrl: string
    label: string
    subLabel: string | JSX.Element
    timeLabel: string | JSX.Element
    hasUnreadMessages: boolean
  }>

const UNREAD_COLOR = color.midnightGreen
const READ_COLOR = color.lightMidnightGreen

const generateSubLabel = (subLabel: string | JSX.Element): JSX.Element => (
  <span className="kirk-messaging-summary-item-sub-label">{subLabel}</span>
)

export const MessagingSummaryItem = ({
  className,
  url,
  pictureUrl,
  label,
  subLabel,
  timeLabel,
  hasUnreadMessages,
  hasHorizontalSpacing = false,
}: MessagingSummaryItemProps) => (
  <StyledMessagingSummaryItem
    className={cc(['kirk-messaging-summary-item', className])}
    leftTitle={label}
    leftTitleDisplay={TextDisplayType.TITLESTRONG}
    leftTitleColor={hasUnreadMessages ? UNREAD_COLOR : READ_COLOR}
    leftBody={generateSubLabel(subLabel)}
    leftBodyDisplay={TextDisplayType.BODYSTRONG}
    leftBodyColor={hasUnreadMessages ? UNREAD_COLOR : READ_COLOR}
    leftBodyAnnotation={timeLabel}
    leftBodyAnnotationDisplay={TextDisplayType.CAPTION}
    rightAddon={<Avatar image={pictureUrl} />}
    chevron
    href={url}
    isClickable
    hasHorizontalSpacing={hasHorizontalSpacing}
  />
)
