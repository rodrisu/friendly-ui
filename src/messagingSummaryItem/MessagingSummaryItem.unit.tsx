import React from 'react'
import renderer from 'react-test-renderer'

import { MessagingSummaryItem } from './MessagingSummaryItem'

const defaultProps = {
  url: 'http_test.url',
  pictureUrl: 'http_test.picture.url',
  label: 'A label',
  subLabel: 'A sub label',
  timeLabel: '3 days ago',
  hasUnreadMessages: true,
}

it('Should render properly with unread messages', () => {
  const item = <MessagingSummaryItem {...defaultProps} />
  const messagingSummaryItem = renderer.create(item).toJSON()
  expect(messagingSummaryItem).toMatchSnapshot()
})

it('Should render properly with read messages', () => {
  const props = {
    ...defaultProps,
    hasUnreadMessages: false,
  }
  const item = <MessagingSummaryItem {...props} />
  const messagingSummaryItem = renderer.create(item).toJSON()
  expect(messagingSummaryItem).toMatchSnapshot()
})
