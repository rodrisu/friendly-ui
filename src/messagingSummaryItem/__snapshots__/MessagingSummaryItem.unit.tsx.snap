// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render properly with read messages 1`] = `
.c3 {
  fill: #708C91;
}

.c3.kirk-icon-wrapper {
  display: inline-block;
  position: relative;
}

.c3.kirk-icon-shadowed {
  border-radius: 50%;
  box-shadow: 0 0 2px rgba(0,0,0,.3);
}

.c3 .kirk-icon-badge {
  position: absolute;
  bottom: 12px;
  left: 12px;
}

.c2 {
  border: 2px dashed #DDD;
  box-sizing: border-box;
  border-radius: 50%;
  position: relative;
  width: 48px;
  height: 48px;
}

.c2.kirk-avatar--small {
  height: 24px;
  width: 24px;
}

.c2.kirk-avatar--medium {
  height: 96px;
  width: 96px;
}

.c2.kirk-avatar--large {
  height: 156px;
  width: 156px;
}

.c2.kirk-avatar--image {
  border: none;
}

.c2.kirk-avatar--image img {
  border-radius: 50%;
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c2 .kirk-badge.kirk-avatar-badge--unreadNotifications,
.c2 .kirk-badge.kirk-avatar-badge--idCheck {
  z-index: 2;
  position: absolute;
}

.c2 .kirk-badge.kirk-avatar-badge--unreadNotifications {
  top: -6px;
  right: -6px;
}

.c2 .kirk-badge.kirk-avatar-badge--idCheck {
  bottom: -3px;
  right: -3px;
  width: 18px;
  height: 18px;
  padding: 0;
  border-radius: 50%;
  background-color: #5DD167;
}

.c2.kirk-avatar--medium .kirk-badge.kirk-avatar-badge--idCheck,
.c2.kirk-avatar--large .kirk-badge.kirk-avatar-badge--idCheck {
  bottom: 0;
  right: 0;
}

.c2 .kirk-avatar--medium .kirk-badge.kirk-avatar-badge--idCheck {
  width: 24px;
  height: 24px;
}

.c2 .kirk-avatar--large .kirk-badge.kirk-avatar-badge--idCheck {
  width: 36px;
  height: 36px;
}

.c2 .kirk-avatar--medium .kirk-badge.kirk-avatar-badge--unreadNotifications {
  height: 24px;
  line-height: 23px;
  font-size: 14px;
  padding: 0 6px;
}

.c2 .kirk-avatar--large .kirk-badge.kirk-avatar-badge--unreadNotifications {
  height: 36px;
  line-height: 34px;
  font-size: 18px;
  padding: 0 8px;
}

.c1 {
  margin: 0;
  font-weight: 400;
}

.c1.kirk-text-button {
  color: #054752;
  font-size: 16px;
  line-height: 20px;
}

.c1.kirk-text-body,
.c1.kirk-text-bodyStrong {
  color: #708C91;
  font-size: 16px;
  line-height: 20px;
}

.c1.kirk-text-caption {
  color: #708C91;
  font-size: 13px;
  line-height: 16px;
}

.c1.kirk-text-title,
.c1.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c1.kirk-text-display2 {
  color: #054752;
  font-size: 82px;
  line-height: 82px;
  font-weight: 500;
}

.c1.kirk-text-display1 {
  color: #054752;
  font-size: 30px;
  line-height: 1.06;
  font-weight: 500;
}

.c1.kirk-text-subheader,
.c1.kirk-text-subheaderStrong {
  color: #054752;
  font-size: 22px;
  line-height: 24px;
}

.c1.kirk-text-title,
.c1.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c1.kirk-text-bodyStrong,
.c1.kirk-text-subheaderStrong,
.c1.kirk-text-titleStrong {
  font-weight: 500;
}

.c0 {
  padding-left: 24px !important;
  padding-right: 24px !important;
  margin-left: 0;
  margin-right: 0;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 0;
  background: none;
  white-space: pre-line;
}

.c0.kirk-item--clickable:hover {
  cursor: pointer;
  background: #EDEDED;
}

.c0.kirk-item--hideHoverBackground.kirk-item--clickable:hover {
  background: none;
}

.c0.kirk-item--wrappable {
  -webkit-flex-flow: wrap;
  -ms-flex-flow: wrap;
  flex-flow: wrap;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c0.kirk-item--wrappable .kirk-item-leftWrapper {
  padding-top: 8px;
  padding-bottom: 8px;
}

.c0.kirk-item--wrappable .kirk-item-leftText {
  margin-right: 24px;
}

.c0.kirk-item--wrappable .kirk-item-rightText {
  margin-left: 0;
  margin-top: 8px;
  margin-bottom: 8px;
  text-align: left;
}

.c0 .kirk-item-tagToken {
  display: block;
  padding-bottom: 4px;
}

.c0:disabled {
  cursor: default;
}

a.c0 {
  background: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button.c0 {
  cursor: pointer;
  text-align: left;
  width: 100%;
  font-family: inherit;
  background-color: transparent;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button:focus:not(.focus-visible).c0 {
  outline: none;
}

.c0 .kirk-item-leftWrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .kirk-item-leftText {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 .kirk-item-rightText {
  margin-left: 16px;
  text-align: right;
}

.c0 .kirk-text {
  display: block;
}

.c0 .kirk-item-title--withBody {
  margin-bottom: 4px;
}

.c0 .kirk-item--strikethrough {
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

.c0 .kirk-item-title--withButtonAddon {
  position: relative;
}

.c0 .kirk-item-leftAddon,
.c0 .kirk-item-rightAddon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .kirk-item-leftAddon {
  margin-right: 16px;
}

.c0 .kirk-item-rightAddon {
  margin-left: 16px;
}

.c0 .kirk-item-rightText--withAddon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .kirk-item-rightText--withAddon svg {
  margin-right: 8px;
}

.c0.kirk-item--highlighted .kirk-text-title {
  color: #00AFF5;
}

.c0 .kirk-item-title--withButtonAddon button {
  position: absolute;
  width: 24px;
  height: 24px;
  margin-left: 4px;
  bottom: 0;
}

.c0 .kirk-messaging-summary-item-sub-label {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  word-break: break-word;
  max-height: 2.5em;
  overflow: hidden;
}

<a
  className="kirk-item kirk-item--clickable c0 kirk-messaging-summary-item"
  href="http_test.url"
>
  <span
    className="kirk-item-leftWrapper"
  >
    <span
      className="kirk-item-leftText"
    >
      
      <span
        className="kirk-text kirk-text-titleStrong c1"
        style={
          Object {
            "color": "#708C91",
          }
        }
      >
        A label
      </span>
      <span
        className="kirk-text kirk-text-bodyStrong c1 kirk-item-body"
        style={
          Object {
            "color": "#708C91",
          }
        }
      >
        <span
          className="kirk-messaging-summary-item-sub-label"
        >
          A sub label
        </span>
      </span>
      <span
        className="kirk-text kirk-text-caption c1 kirk-item-body-annotation"
      >
        3 days ago
      </span>
    </span>
  </span>
  <span
    className="kirk-item-rightAddon"
  >
    <div
      className="c2 kirk-avatar--image kirk-avatar"
    >
      <img
        alt=""
        src="http_test.picture.url"
      />
    </div>
  </span>
  <span
    className="kirk-item-rightAddon"
  >
    <svg
      aria-hidden={true}
      className="kirk-icon c3"
      height={24}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      
      <polyline
        fill="none"
        points="9 18 15 12 9 6"
        stroke="#708C91"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeMiterlimit="10"
        strokeWidth="2"
      />
    </svg>
  </span>
</a>
`;

exports[`Should render properly with unread messages 1`] = `
.c3 {
  fill: #708C91;
}

.c3.kirk-icon-wrapper {
  display: inline-block;
  position: relative;
}

.c3.kirk-icon-shadowed {
  border-radius: 50%;
  box-shadow: 0 0 2px rgba(0,0,0,.3);
}

.c3 .kirk-icon-badge {
  position: absolute;
  bottom: 12px;
  left: 12px;
}

.c2 {
  border: 2px dashed #DDD;
  box-sizing: border-box;
  border-radius: 50%;
  position: relative;
  width: 48px;
  height: 48px;
}

.c2.kirk-avatar--small {
  height: 24px;
  width: 24px;
}

.c2.kirk-avatar--medium {
  height: 96px;
  width: 96px;
}

.c2.kirk-avatar--large {
  height: 156px;
  width: 156px;
}

.c2.kirk-avatar--image {
  border: none;
}

.c2.kirk-avatar--image img {
  border-radius: 50%;
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c2 .kirk-badge.kirk-avatar-badge--unreadNotifications,
.c2 .kirk-badge.kirk-avatar-badge--idCheck {
  z-index: 2;
  position: absolute;
}

.c2 .kirk-badge.kirk-avatar-badge--unreadNotifications {
  top: -6px;
  right: -6px;
}

.c2 .kirk-badge.kirk-avatar-badge--idCheck {
  bottom: -3px;
  right: -3px;
  width: 18px;
  height: 18px;
  padding: 0;
  border-radius: 50%;
  background-color: #5DD167;
}

.c2.kirk-avatar--medium .kirk-badge.kirk-avatar-badge--idCheck,
.c2.kirk-avatar--large .kirk-badge.kirk-avatar-badge--idCheck {
  bottom: 0;
  right: 0;
}

.c2 .kirk-avatar--medium .kirk-badge.kirk-avatar-badge--idCheck {
  width: 24px;
  height: 24px;
}

.c2 .kirk-avatar--large .kirk-badge.kirk-avatar-badge--idCheck {
  width: 36px;
  height: 36px;
}

.c2 .kirk-avatar--medium .kirk-badge.kirk-avatar-badge--unreadNotifications {
  height: 24px;
  line-height: 23px;
  font-size: 14px;
  padding: 0 6px;
}

.c2 .kirk-avatar--large .kirk-badge.kirk-avatar-badge--unreadNotifications {
  height: 36px;
  line-height: 34px;
  font-size: 18px;
  padding: 0 8px;
}

.c1 {
  margin: 0;
  font-weight: 400;
}

.c1.kirk-text-button {
  color: #054752;
  font-size: 16px;
  line-height: 20px;
}

.c1.kirk-text-body,
.c1.kirk-text-bodyStrong {
  color: #708C91;
  font-size: 16px;
  line-height: 20px;
}

.c1.kirk-text-caption {
  color: #708C91;
  font-size: 13px;
  line-height: 16px;
}

.c1.kirk-text-title,
.c1.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c1.kirk-text-display2 {
  color: #054752;
  font-size: 82px;
  line-height: 82px;
  font-weight: 500;
}

.c1.kirk-text-display1 {
  color: #054752;
  font-size: 30px;
  line-height: 1.06;
  font-weight: 500;
}

.c1.kirk-text-subheader,
.c1.kirk-text-subheaderStrong {
  color: #054752;
  font-size: 22px;
  line-height: 24px;
}

.c1.kirk-text-title,
.c1.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c1.kirk-text-bodyStrong,
.c1.kirk-text-subheaderStrong,
.c1.kirk-text-titleStrong {
  font-weight: 500;
}

.c0 {
  padding-left: 24px !important;
  padding-right: 24px !important;
  margin-left: 0;
  margin-right: 0;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 0;
  background: none;
  white-space: pre-line;
}

.c0.kirk-item--clickable:hover {
  cursor: pointer;
  background: #EDEDED;
}

.c0.kirk-item--hideHoverBackground.kirk-item--clickable:hover {
  background: none;
}

.c0.kirk-item--wrappable {
  -webkit-flex-flow: wrap;
  -ms-flex-flow: wrap;
  flex-flow: wrap;
  padding-top: 8px;
  padding-bottom: 8px;
}

.c0.kirk-item--wrappable .kirk-item-leftWrapper {
  padding-top: 8px;
  padding-bottom: 8px;
}

.c0.kirk-item--wrappable .kirk-item-leftText {
  margin-right: 24px;
}

.c0.kirk-item--wrappable .kirk-item-rightText {
  margin-left: 0;
  margin-top: 8px;
  margin-bottom: 8px;
  text-align: left;
}

.c0 .kirk-item-tagToken {
  display: block;
  padding-bottom: 4px;
}

.c0:disabled {
  cursor: default;
}

a.c0 {
  background: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button.c0 {
  cursor: pointer;
  text-align: left;
  width: 100%;
  font-family: inherit;
  background-color: transparent;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button:focus:not(.focus-visible).c0 {
  outline: none;
}

.c0 .kirk-item-leftWrapper {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .kirk-item-leftText {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c0 .kirk-item-rightText {
  margin-left: 16px;
  text-align: right;
}

.c0 .kirk-text {
  display: block;
}

.c0 .kirk-item-title--withBody {
  margin-bottom: 4px;
}

.c0 .kirk-item--strikethrough {
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

.c0 .kirk-item-title--withButtonAddon {
  position: relative;
}

.c0 .kirk-item-leftAddon,
.c0 .kirk-item-rightAddon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .kirk-item-leftAddon {
  margin-right: 16px;
}

.c0 .kirk-item-rightAddon {
  margin-left: 16px;
}

.c0 .kirk-item-rightText--withAddon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .kirk-item-rightText--withAddon svg {
  margin-right: 8px;
}

.c0.kirk-item--highlighted .kirk-text-title {
  color: #00AFF5;
}

.c0 .kirk-item-title--withButtonAddon button {
  position: absolute;
  width: 24px;
  height: 24px;
  margin-left: 4px;
  bottom: 0;
}

.c0 .kirk-messaging-summary-item-sub-label {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  word-break: break-word;
  max-height: 2.5em;
  overflow: hidden;
}

<a
  className="kirk-item kirk-item--clickable c0 kirk-messaging-summary-item"
  href="http_test.url"
>
  <span
    className="kirk-item-leftWrapper"
  >
    <span
      className="kirk-item-leftText"
    >
      
      <span
        className="kirk-text kirk-text-titleStrong c1"
        style={
          Object {
            "color": "#054752",
          }
        }
      >
        A label
      </span>
      <span
        className="kirk-text kirk-text-bodyStrong c1 kirk-item-body"
        style={
          Object {
            "color": "#054752",
          }
        }
      >
        <span
          className="kirk-messaging-summary-item-sub-label"
        >
          A sub label
        </span>
      </span>
      <span
        className="kirk-text kirk-text-caption c1 kirk-item-body-annotation"
      >
        3 days ago
      </span>
    </span>
  </span>
  <span
    className="kirk-item-rightAddon"
  >
    <div
      className="c2 kirk-avatar--image kirk-avatar"
    >
      <img
        alt=""
        src="http_test.picture.url"
      />
    </div>
  </span>
  <span
    className="kirk-item-rightAddon"
  >
    <svg
      aria-hidden={true}
      className="kirk-icon c3"
      height={24}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      
      <polyline
        fill="none"
        points="9 18 15 12 9 6"
        stroke="#708C91"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeMiterlimit="10"
        strokeWidth="2"
      />
    </svg>
  </span>
</a>
`;
